### DISCUSSION TITLE
How to dynamically update global styles

### DISCUSSION THREAD

**Discussion by oahmaro** [chakra-ui/chakra-ui#3146] [Q&A] [answered]
I am trying to implement a feature where users can choose to underline all links in the app to help them distinguish links easily. and i would like to know what feature in chakra-ui is best suited for this. and is it possible to do this dynamically?

**Comment by dodas** [comment, marked as answer]
hey @oahmaro

I would use Chakra's [global styles](https://chakra-ui.com/docs/features/global-styles) with a classname, something like this:
```
export const theme = extendTheme({
  styles: {
    global: {
      ".distinguish-links a": {
        textDecoration: "underline",
      }
    }
  }
})
```

And then conditionally add a `distinguish-links` class to any wrapper element.
You could use similar technique inside component theme as well, for example to override it only for `Link` component, not all `a`..
Working example: https://codesandbox.io/s/extend-variants-demo-forked-xlcjg?file=/src/theme/index.ts

**Reply by moutikabdessabour** [reply]
@dodas I'm trying to achieve something similar (add a class style to a button) but only affect one `variant` of the `Button` component. is this possible in chakra? Thank you
